### ISSUE TITLE
Documentation request: WinAVR and Arduino IDE

### ISSUE THREAD

**Issue by sorjon02** [MarlinFirmware/Marlin#14531]
I am trying to set up the Arduino IDE to compile Marlin 2.0. I understand that this is a prerelease, and still under development, But I am willing to accept this, to have a look at the new version and maybe be able to contribute something to the Project.
Working computer for development: PC; OS: Microsoft Windows 10

So far I have:

1. Downloaded and installed Arduino IDE 1.8.9 C:\Users\[UserName]\Documents\Arduino is the 
Schetch dir.

2. Downloaded and installed WinAVR  (WinAVR-20100110-install) in c:\Users\[UserName]\Documents\Arduino\libraries\WinAVR

3. Downloaded support for Arduino Due and configured the Project for this CPU

4. Downloaded Marlin using "git clone https://github.com/MarlinFirmware/Marlin.git" to c:\Users\[UserName]\Documents\Arduino\Marlin

5. Attempted to compile Marlin 2.0, but with the error message "exit status 1
avr/io.h: No such file or directory" This is why I installed WinAVR

Despite my efforts during today, I have not been able to make the WinAVR Library available in the Arduino IDE. And yes: I have spent my time serching the Internet.

Can somebodey please spend a Little time documenting how to do this? If I can get it right and compile Marln 2.0 I'm willing to write a script to make the Marlin download and configure for compilation process automatically. (I have already began a script in PowerShell, alltrough so far it only checks that the correct versions of some SW is available.)

**Comment by sl1pkn07** [comment]
why you not try to use platformio?

**Comment by reloxx13** [comment]
no need for winavr

its included in arduino ide

move ur arduino folder to root of C: (arduino ide always had some path issues, eg. special chars/spaces in username dir)   
also move marlin to a clean folder path like c:\projects\marlin
    
![grafik](https://user-images.githubusercontent.com/14855001/60758751-26619000-a01b-11e9-9bbc-85cc7e657c94.png)

2.0 compiled fine

dont use windows appstore arduino ide,
use latest arduino IDE for marlin 2 (cause of subfolders)

![grafik](https://user-images.githubusercontent.com/14855001/60758783-8b1cea80-a01b-11e9-8368-0f32c0de8912.png)

![grafik](https://user-images.githubusercontent.com/14855001/60758796-a8ea4f80-a01b-11e9-84a4-5a7ff8aab557.png)

(ender 3 board)
![grafik](https://user-images.githubusercontent.com/14855001/60758820-e4851980-a01b-11e9-8c5b-51860a38233f.png)


ps: platformio is the way to go, arduino ide sucks for big projects.

**Comment by sorjon02** [comment]
Problem solved! Arduino IDE is built on an early version of Gnu AVR/WinAVR, and does not support the avr/io.h include file and library.  My guess is that upgrading the support requires a new major version, and is probably not worth the effort, especially if your main interest is Marlin.

The best solution for you is probably to drop all support for Arduino IDE, and only support Atom/PlatformIO. If you state this on the website, it will save anyone interested in compiling Marlin 2 a lot of headscratching and effort.

**Comment by reloxx13** [comment]
dude, i just compiled marlin 2.0, i even posted screenshots of it
did u even read it?

**Comment by sorjon02** [comment]
I did read it! But I also spent two or three hours looking for the library avr/io.h in the Arduino IDE directory and subdirectories. Despite my effort, I could not find it. I decided to give up, and go for PlatformIO instead, and Marlin 2 compiled imidiately. So I go for Atom/PlatformIO now. I don't want to spend more time trying to make Arduino IDE work for me.

**Comment by thinkyhead** [comment]
Atom is no longer being supported by PlatformIO.
Going forward you should all use Visual Studio Code instead.
MarlinFirmware/MarlinDocumentation#176

**Comment by sorjon02** [comment]
I have now changed from Atom/PlatformIO to Visual Studio/PlatformIO, in accordance with  "Switch to PlatformIO IDE for VSCode #176 ". Sorry for this, but there has been a lot of invormation to take in, and I feel that it is fragmentetd. When it comes to Marlin development, it feels like everytime I has been presented with two choises I have managed to take the worst one. It's bad luck, but that's Life sometimes.
Best regards,
